pandas/util/_exceptions.py

```diff
@@ -25,20 +26,24 @@ def rewrite_exception(old_name: str, new_name: str) -> Iterator[None]:
         raise
 
 
-def find_stack_level() -> int:
+@functools.lru_cache
+def find_stack_level(frame) -> int:
     """
     Find the first place in the stack that is not inside pandas
     (tests notwithstanding).
+
+    ``frame`` should be passed as ``inspect.currentframe()`` by the
+    calling function.
+
+    https://stackoverflow.com/questions/17407119/python-inspect-stack-is-slow
     """
 
     import pandas as pd
 
     pkg_dir = os.path.dirname(pd.__file__)
     test_dir = os.path.join(pkg_dir, "tests")
 
-    # https://stackoverflow.com/questions/17407119/python-inspect-stack-is-slow
-    frame = inspect.currentframe()
-    n = 0
+    n = 1
```

Comment:
Curious why this is 1 instead of 0 now.

Reply:
Because the location where `inspect.currentframe()` is moved to where `find_stack_level` is called, rather than within `find_stack_level` itself

Reply:
Gotcha thanks!